core/drivers/stm32_gpio.c

```diff
@@ -393,6 +397,67 @@ static struct stm32_gpio_bank *stm32_gpio_get_bank(unsigned int bank_id)
 	panic();
 }
 
+#if defined(CFG_STM32_RIF)
+static bool pin_is_accessible(struct stm32_gpio_bank *bank, unsigned int pin)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	bool accessible = false;
+	uint32_t cidcfgr = 0;
+
+	if (!bank->rif_cfg)
+		return true;
+
+	if (clk_enable(bank->clock))
+		panic();
+
+	cidcfgr = io_read32(bank->base + GPIO_CIDCFGR(pin));
+
+	if (!(cidcfgr & _CIDCFGR_CFEN)) {
+		/* Resource can be accessed when CID filtering is disabled */
+		accessible = true;
+	} else if (stm32_rif_scid_ok(cidcfgr, GPIO_CIDCFGR_SCID_MASK,
+				     RIF_CID1)) {
+		/* Resource can be accessed if CID1 is statically allowed */
+		accessible = true;
+	} else if (stm32_rif_semaphore_enabled_and_ok(cidcfgr, RIF_CID1)) {
+		/* We must acquire the semaphore to access the resource */
```

Comment:
We should add a stm32_rif_semaphore_is_available_or_taken() to check if the current CID has already taken the semaphore. This will avoid some issue when acquiring the semaphore (spurious IAC). Maybe we can do this in another P-R. (I can take care of that)

Reply:
To prevent such issues when acquiring an already taken semaphore, I think we should rather fix `stm32_rif_acquire_semaphore()`:
```patch
 TEE_Result stm32_rif_acquire_semaphore(vaddr_t addr, unsigned int nb_cid_supp)
 {
 	uint32_t scid_mask = get_scid_mask(nb_cid_supp);
 
-	/* Take the semaphore */
-	io_setbits32(addr, _SEMCR_MUTEX);
+	/* Take the semaphore is not already taken */
+	if (stm32_rif_semaphore_is_available(addr))
+		io_setbits32(addr, _SEMCR_MUTEX);

	/* Check that the Cortex-A has the semaphore */
	if (stm32_rif_semaphore_is_available(addr) ||
	    ((io_read32(addr) & scid_mask) >> _CIDCFGR_SCID_SHIFT) != RIF_CID1)
		return TEE_ERROR_ACCESS_DENIED;

	return TEE_SUCCESS;
}